plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -705,7 +705,7 @@ public ExecutionResult createFileInVR(String routerIp, String filePath, String f
         VmwareManager mgr = getServiceContext().getStockObject(VmwareManager.CONTEXT_STOCK_NAME);
         File keyFile = mgr.getSystemVMKeyFile();
         try {
-            SshHelper.scpTo(routerIp, 3922, "root", keyFile, null, filePath, content.getBytes(), fileName, null);
+            SshHelper.scpTo(routerIp, 3922, "root", keyFile, null, filePath, content.getBytes("UTF-8"), fileName, null);
```

Comment:
VR runs on a UTF-8 platform.
If you're running this on windows and output a specific set of bytes based on windows' encoding, the result will be different when VR reads the file. UTF-8 should be used specifically in this case, since we're outputting a specific array of bytes to a platform with known encoding. Charset.defaultCharset() would only work well if the platorm it is running on matches that encoding which it actually is in most cases since most peeps run this on a modern linux distro ;)